src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1508,6 +1508,18 @@
     <Field Name="WasTargetTyped" Type="bool" />
   </Node>
 
+  <Node Name="BoundLoweredSwitchExpression" Base="BoundExpression" HasValidate="true">
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <Field Name="Statements" Type="ImmutableArray&lt;BoundStatement&gt;"/>
+    <Field Name="SwitchArms" Type="ImmutableArray&lt;BoundLoweredSwitchExpressionArm&gt;"/>
+  </Node>
+
+  <Node Name="BoundLoweredSwitchExpressionArm" Base="BoundNode">
+    <Field Name="Locals" Type="ImmutableArray&lt;LocalSymbol&gt;"/>
```

Comment:
><Field Name="Locals" Type="ImmutableArray&lt;LocalSymbol&gt;"/> [](http://example.com/codeflow?start=4&length=63)

I think it will be preferred to not introduce new nodes with locals that survive lowering. Can we somehow represent what we want with combination of existing nodes?